include/envoy/secret/secret.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "envoy/common/exception.h"
+#include "envoy/ssl/context.h"
+
+namespace Envoy {
+namespace Secret {
+
+/**
+ * Secret contains certificate chain and private key
+ */
+class Secret {
+public:
+  virtual ~Secret() {}
+
+  /**
+   * @return a name of the SDS secret
+   */
+  virtual const std::string& getName() PURE;
```

Comment:
make these getters as const method?

Also I would prefer not having `get` prefix for those properties, so just `name`, `certificateChain`...

Reply:
Done